src/Nethermind/Nethermind.JsonRpc/Modules/Proof/ProofModule.cs

```diff
@@ -0,0 +1,213 @@
+    {
+        private readonly ILogger _logger;
+        private readonly ITracer _tracer;
+        private readonly IBlockFinder _blockFinder;
+        private readonly IReceiptFinder _receiptFinder;
+        private readonly ISpecProvider _specProvider;
+        private readonly HeaderDecoder _headerDecoder = new HeaderDecoder();
+
+        public ProofModule(
+            ITracer tracer,
+            IBlockFinder blockFinder,
+            IReceiptFinder receiptFinder,
+            ISpecProvider specProvider,
+            ILogManager logManager)
+        {
+            _tracer = tracer ?? throw new ArgumentNullException(nameof(tracer));
+            _blockFinder = blockFinder ?? throw new ArgumentNullException(nameof(blockFinder));
+            _receiptFinder = receiptFinder ?? throw new ArgumentNullException(nameof(receiptFinder));
+            _specProvider = specProvider ?? throw new ArgumentNullException(nameof(specProvider));
+            _logger = logManager?.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
+        }
+
+        [Todo(Improve.Review, "Double check the way the trace header is onstructed based on the block parameter. SHall we assign exactly the same hash / difficulty and other properties?")]
+        public ResultWrapper<CallResultWithProof> proof_call(TransactionForRpc tx, BlockParameter blockParameter)
+        {
+            Transaction transaction = tx.ToTransaction();
+
+            BlockHeader header;
+            if (blockParameter.RequireCanonical)
+            {
+                header = _blockFinder.FindHeader(blockParameter.BlockHash, BlockTreeLookupOptions.RequireCanonical);
+                if (header == null)
+                {
+                    header = _blockFinder.FindHeader(blockParameter);
+                    if (header != null)
+                    {
+                        return ResultWrapper<CallResultWithProof>.Fail($"{blockParameter.BlockHash} block is not canonical", ErrorCodes.InvalidInput);
+                    }
+                }
+            }
+            else
+            {
+                header = _blockFinder.FindHeader(blockParameter);
+            }
+
+            if (header == null)
+            {
+                return ResultWrapper<CallResultWithProof>.Fail($"{blockParameter.BlockHash} could not be found", ErrorCodes.ResourceNotFound);
+            }
+
+            Block block = new Block(header, new[] {transaction}, Enumerable.Empty<BlockHeader>());
+            block.Author = Address.Zero;
+            block.TotalDifficulty = 0; // not used in EVM
+
+            ProofBlockTracer proofBlockTracer = new ProofBlockTracer(null);
+            _tracer.Trace(block, proofBlockTracer);
+
+            CallResultWithProof callResultWithProof = new CallResultWithProof();
+            ProofTxTracer proofTxTracer = proofBlockTracer.BuildResult().Single();
+
+            
+            
+            CollectHeaders(proofTxTracer, callResultWithProof);
```

Comment:
Would change the method to return result and usage to 
callResultWithProof.BlockHeaders = CollectHeaders(proofTxTracer);
would be more readable. Its like that for CollectAccountProofs

Reply:
ok, initally both were the other way, will change it